app.py

```diff
@@ -244,12 +245,24 @@ def internal_server_error(e):
 def handle_postback():
     today_date = datetime.now().strftime("%m/%d/%Y %H:%M")
     citybase_data = request.get_json()
-    # TODO: validate citybase_data
     # information from citybase payload
     app.logger.info(f"New POST with payload: {citybase_data}")
+
+    try:
+        validate(citybase_data, payment_reporting_schema)
+    except ValidationError as e:
+        app.logger.error(f"Validation error: {e}")
+        return f"Validation error: {e.message}", 400
```

Comment:
I like the `400 BAD REQUEST` choice here -- that's perfect. I was wondering if there were best practices that anyone would recommend around what the payload could look like, and it turns out that there is a proposed standard working it's way through the IETF that suggests that the payload on a `400` response should be machine parsable, which may want to consider. I think something that emits something like the following would totally do it.

```
{
   "error_type": "Validation Failure", 
   "error_message": "{e.message}"
}
```